subsys/bluetooth/services/bas_c.c

```diff
@@ -272,3 +388,38 @@ int bt_gatt_bas_c_get(struct bt_gatt_bas_c *bas_c)
 	}
 	return bas_c->battery_level;
 }
+
+
+int bt_gatt_bas_c_periodic_read_start(struct bt_gatt_bas_c *bas_c,
+				      s32_t interval,
+				      bt_gatt_bas_c_read_cb func)
+{
+	if (!bas_c || !func) {
+		return -EINVAL;
+	}
+
+	if (bt_gatt_bas_c_notify_supported(bas_c)) {
+		return -ENOTSUP;
+	}
+
```

Comment:
Please handle the case when the second start is triggered when the read process is already running. Decide what you think is best:
- don't allow restart without ```bt_gatt_bas_c_periodic_read_stop``` - return error
- handle it internally (call ```bt_gatt_bas_c_periodic_read_stop```) and proceed with the starting logic

Reply:
I resolved it in a little different kind. Please look at it.